### PR TITLE
feature/edit-watchlist

### PR DESCRIPTION
**Pull request by haritonasty**
1) Added empty watchlist section
2) Added edit watchlist popup

image

image

image

### CODE REVIEW

File: src/components/WatchlistEdit/WatchlistEdit.js [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+          {editWatchlistState ? 'Applying...' : 'Apply'}
+        </Dialog.Approve>
+      </Dialog.Actions>
+    </Dialog>
+  )
+}
+
+const ROW_HEIGHT = 32
+
+const AssetsList = ({ items, listItems, isContained, onToggleProject }) => {
```
Comment (DmitriVanGuard) [inline src/components/WatchlistEdit/WatchlistEdit.js]: Maybe better to extract this component into a separate file?

File: src/components/Search/SearchProjects.js [hunk 2]
```diff
@@ -19,17 +25,28 @@ const SearchProjects = ({ data: { allProjects = [] }, ...props }) => {
           name.toUpperCase().includes(upperCaseSearchTerm) ||
           ticker.toUpperCase().includes(upperCaseSearchTerm)
       }}
-      suggestionContent={({ name, ticker }) => {
+      suggestionContent={({ name, ticker, id }) => {
+        const isAssetInList = isEditingWatchlist
```
Comment (Partysun) [inline src/components/Search/SearchProjects.js]: @haritonasty isEditi && hasAssetById

File: src/components/WatchlistEdit/editAssetsInWatchlistEpic.js [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+        }
+      })
+      return Observable.from(userListUpdate)
+        .mergeMap(() =>
+          Observable.of({
+            type: actions.USER_EDIT_ASSETS_IN_LIST_SUCCESS,
+            payload: { listItems, assetsListId }
+          })
+        )
+        .catch(error => {
```
Comment (Partysun) [inline src/components/WatchlistEdit/editAssetsInWatchlistEpic.js]: We have handleErrorAndTriggerAction util method for this.
@haritonasty

File: src/components/WatchlistEdit/editAssetsInWatchlistEpic.js [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+export const addAssetToWatchlistEpic = (action$, store, { client }) =>
+  action$
+    .ofType(actions.USER_ADD_ASSET_TO_LIST)
+    .mergeMap(({ payload: { assetsListId, listItems = [], projectId } }) => {
+      const normalizedList = listItems.map(val => ({ project_id: +val.id }))
+      const newListItems = projectId
+        ? [...normalizedList, { project_id: +projectId }]
+        : normalizedList
+
+      const userListUpdate = client.mutate({
```
Comment (Partysun) [inline src/components/WatchlistEdit/editAssetsInWatchlistEpic.js]: Is it the same code, as above? @haritonasty

**Pull request merged**